src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -950,6 +951,27 @@ public ResponseEntity<List<User>> searchOtherUsersInCourse(@PathVariable long co
         return ResponseEntity.ok().body(courseService.searchOtherUsersNameInCourse(course, nameOfUser));
     }
 
+    /**
+     * GET /api/courses/:courseId/members/search: Searches for members of a course
+     *
+     * @param courseId    id of the course
+     * @param loginOrName the search term to search login and names by
+     * @return the ResponseEntity with status 200 (OK) and with body containing the list of found members matching the criteria
+     */
+    @GetMapping("courses/{courseId}/members/search")
+    @EnforceAtLeastStudent
+    public ResponseEntity<List<UserNameAndLoginDTO>> searchMembersOfCourse(@PathVariable Long courseId, @RequestParam("loginOrName") String loginOrName) {
+        log.debug("REST request to get members with login or name : {} in course: {}", loginOrName, courseId);
+
+        var course = courseRepository.findByIdElseThrow(courseId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);
+
+        var searchTerm = loginOrName != null ? loginOrName.toLowerCase().trim() : "";
+        List<User> searchResults = userRepository.searchAllByLoginOrNameInCourse(Pageable.ofSize(10), searchTerm, course.getId()).toList();
```

Comment:
Is it possible to directly map or stream the page instead of first converting it into a list just to convert it to a stream in the very next line?

Reply:
I changed it. Good catch, thank you